app/models/user.rb

```diff
@@ -153,20 +153,16 @@ def has_tag(tagname)
 
    # power tags have "key:value" format, and should be searched with a "key:*" wildcard
   def has_power_tag(key)
-    all_key_tags_ids = Tag.where('name LIKE ?' , key + ':%').collect(&:tid)
-    tids = TagSelection.where('user_id = ? AND tid IN (?)', self.id , all_key_tags_ids)
-    !tids.blank?
+     tids = user_tags.where('value LIKE ?' , key + ':%').collect(&:id)
```

Comment:
I like adding `self.` just to make it clear where this reference goes. Would you mind changing that? Thank you!!!